lib/mas/firm_indexer.rb

```diff
@@ -0,0 +1,32 @@
+module FirmIndexer
+  class << self
+    def index_firm(firm)
+      if !firm.destroyed? && firm.publishable?
+        store_firm(firm)
+      else
+        delete_firm(firm)
+      end
+    end
+
+    def handle_firm_changed(firm)
+      index_firm(firm)
+    end
+
+    def handle_aggregate_changed(aggregate)
+      index_firm(aggregate.firm) if firm_exists?(aggregate.firm)
```

Comment:
Would it be beneficial to make this conditional a guard in `index_firm` instead? It is currently possible to pass something that is not an existing Firm to that method.